demo/dismissable/dismissable-item.js

```diff
@@ -0,0 +1,219 @@
+        break;
+    }
+  }
+
+  setPosition(position) {
+    this.position = position;
+    this.width = this.offsetWidth;
+    this.style.opacity = (this.width - Math.abs(position)) / this.width;
+    const currentY = this.style.transform.split(',')[1];
+    this.style.transform = `translate(${position}px, ${currentY}`;
+  }
+
+  _dismiss() {
+    this.style.opacity = 0;
+
+    const collapseAnim = this.animate({
+      height: [ getComputedStyle(this).height, '0px']
+    }, {
+      duration: 100,
+      iterations: 1
+    });
+
+    collapseAnim.onfinish = this._fireRemove.bind(this);
+  }
+
+  _fireRemove() {
+    this.setPosition(0);
+
+    const event = new CustomEvent('remove', {
+      detail: { itemIndex: this.itemIndex },
+      bubbles: true
+    });
+    this.dispatchEvent(event);
+  }
+
+  settle(targetPosition) {
+    this.state = 'initial';
+    if (targetPosition === this.position) {
+      return;
+    }
+
+    const currentY = this.style.transform.split(',')[1];
+    const isDismiss = targetPosition !== 0;
+
+    const animation = this.animate({
+      transform: [
+        `translate(${this.position}px,${currentY}`,
+        `translate(${targetPosition}px,${currentY}`
+      ],
+      opacity: [ this.style.opacity, isDismiss ? 0 : 1 ]
+    }, {
+      duration: Math.abs(targetPosition - this.position) * 0.5,
+      iterations: 1
+    });
+
+    this.position = targetPosition;
+    animation.onfinish = () => isDismiss ? this._dismiss() : this.setPosition(0);
+  }
+
+  fling(velocityX) {
+    this.state = 'initial';
+    const targetPosition = velocityX < 0 ? -this.width : this.width;
+
+    const currentY = this.style.transform.split(',')[1];
+
+    const animation = this.animate({
+      transform: [
+        `translate(${this.position}px,${currentY}`,
+        `translate(${targetPosition}px,${currentY}`
+      ],
+      opacity: [ this.style.opacity, 0 ]
+    }, {
+      duration: Math.abs(targetPosition - this.position) / Math.abs(velocityX),
+      iterations: 1
+    });
+
+    animation.onfinish = this._dismiss.bind(this);
+  }
+
+  _settleToClosestPosition() {
+    const fraction = this.position / this.width;
+    if (fraction > 0.5) {
+      this.settle(this.width);
+    } else if (fraction < -0.5) {
+      this.settle(-this.width);
+    } else {
+      this.settle(0);
+    }
+  }
+
+  _onPointerDown(e, change) {
+    this.state = 'initial';
+    this.startX = change.clientX;
+    this.startY = change.clientY;
+    this.startPosition = 0;
+  }
+
+  _onPointerPan(e, change) {
+    if (this.state == 'initial') {
+      const deltaX = change.clientX - this.startX;
+      const deltaY = change.clientY - this.startY;
+
+      // Shaky or sloppy fingers are common using scroll, so
+      // we ignore mini pans of 5 dips like Android.
+      if (deltaX ** 2 + deltaY ** 2 < kTouchSlopValue ** 2) {
+        return;
+      }
+
+      if (Math.abs(deltaY) > Math.abs(deltaX)) {
+        this._settleToClosestPosition();
```

Comment:
oh! @kenchris I think all we need to do to cancel the dismissal is to set a new state in here, since in this case we captured the fact that there is more vertical scroll than horizontal.
Adding a line here should do the trick:
```js
this._settleToClosestPosition();
this.state = 'cancel';
```
The rest of the code already handles the resetting of the state, and will do position/opacity if state == 'start' or state == 'dragging' only 👌

Reply:
This still doesn't solve that you can sometimes start a horizontal pan after starting to scroll:

https://drive.google.com/file/d/17wW9SmR9WlxwN9uqfvFf9xAvIibr_DZe/view

Also as you see the `overflow-x: hidden` trick doesn't seem to be enough

Reply:
Actually you might be right, it seems that the change maybe wasn't reloaded properly. Now I cannot seem to trigger it anymore \o/

Reply:
No I can still trigger it, by quickly moving down or up in a 90 degree angle - that still start list scrolling and panning

Reply:
If I readd that e.preventDefault() inside the dragging/panning code, then no scroll can happen at the same time that that solves this. Can I re-add that?

Reply:
Argh :x 
This needs more investigation to be perfected. 
I'll have some time to look more into it later, I was thinking to see if keeping only touch* events solves the problem.

Reply:
@valdrinkoshi great. I will upload the preventDefault() fix as that is all working. If you end up fixing it in any other way that is great with me :-)